.github/workflows/e2e-test-provider-example.yml

```diff
@@ -444,4 +449,8 @@ jobs:
         with:
           projectWriteToken: ${{ secrets.PROJECT_WRITE_TOKEN }}
           test: "terraform-provider-example"
+          refStream: ${{ inputs.ref}}
           provider: ${{ inputs.cloudProvider }}
+          kubernetesVersion: ${{ steps.kubernetes.outputs.version }}
+          clusterCreation: "terraform"
+          attestationVariant: ${{ inputs.cloudProvider != 'gcp' && format('{0}-sev-snp', inputs.cloudProvider) || 'gcp-sev-es' }}
```

Comment:
This won't be working for Azure-TDX. If we want to support this in the example test, we should pass it here

Reply:
I made it an input parameter.

Reply:
`attestationVariant` should replace `cloudProvider` as an input parameter, since there are many combinations of the two which result in invalid test setups.

You can look at the manual e2e test code for how we derive `cloudProvider` from `attestationVariant` input.

Eventually this test should support multiple attestationVariants for one CSP, but for this PR its ok to limit the runs to `azure-sev-snp`, `aws-sev-snp` and `gcp-sev-es`.